### PR TITLE
Add a global redjubjub verification context with fallback.

### PR DESCRIPTION
**Pull request by hdevalence**
This moves the `redjubjub` module into a new `primitives` module to prepare for adding a groth16 verifier, and provides a global verification service wrapped in a `once_cell::Lazy`.

### CODE REVIEW

**Comment by hdevalence** [comment]
Clippy warns on unused statics because we don't ever do any verification of redjubjub signatures.

**Comment by teor2345** [comment]
> Clippy warns on unused statics because we don't ever do any verification of redjubjub signatures.

Can we add `allow(dead_code)` in the appropriate place, until these verifiers are used?

Otherwise, we'll get a bunch of spurious clippy warnings on every build and PR after merge.

**Review by teor2345 — DISMISSED** [review]
This is a good design to increase batch sizes.

Do we need to do this with other batch verifiers as well?

Let's allow the dead code for now, so we don't see spurious clippy warnings every time we build.

**Comment by hdevalence** [comment]
Yes, I did this as a small piece of preparatory work for adding a `primitives::groth16` module.  The constants are in the parent module so they can be reused there.

To handle the dead code warnings I would suggest adding an `#[allow(dead_code)]` to the entire `primitives` module for now.

File: zebra-consensus/src/lib.rs
Comment (yaahc) [inline zebra-consensus/src/lib.rs]: TIOLI: I'd prefer this to be an inner attribute on the primitives module so its closer to the code it annotates.

File: zebra-consensus/src/primitives/redjubjub.rs
Comment (yaahc) [inline zebra-consensus/src/primitives/redjubjub.rs:31]: :heart:

File: zebra-consensus/src/primitives/redjubjub.rs
Comment (yaahc) [inline zebra-consensus/src/primitives/redjubjub.rs:30]: Can we make this static private inside of a pub function that does the cloning for users? That way people just call `get_service()` or w/e to get a copy and they can't fuck it up on accident.

**Review by yaahc — CHANGES_REQUESTED** [review]
Looks good, I'd like to see the change to expose the static via a function rather than directly exporting the static but other than that this is PG

File: zebra-consensus/src/primitives/redjubjub.rs
Comment (hdevalence) [reply zebra-consensus/src/primitives/redjubjub.rs:30]: There's no way for them to screw it up because the static is immutable, so the only way to use it is by calling `.clone()`.  The note is just to avoid confusion and point them in the right direction.

Comment (yaahc) [reply zebra-consensus/src/primitives/redjubjub.rs:30]: Aah, yea, for some reason I thought there was a lock. I think it might still be better to point them in the right direction by exposing an interface that encourages the needed pattern rather than explaining it with a comment but I'll leave it up to you.

File: zebra-consensus/src/lib.rs
Comment (hdevalence) [reply zebra-consensus/src/lib.rs]: I put it here so it sticks out more and we'll be less likely to forget to remove it later.

Comment (hdevalence) [reply zebra-consensus/src/lib.rs]:
```suggestion
#[allow(dead_code)] // Remove this once transaction verification is implemented
```
WDYT? @yaahc

Comment (yaahc) [reply zebra-consensus/src/lib.rs]: sounds good

**Pull request merged**
